muswarmadmin/__main__.py

```diff
@@ -1,28 +1,38 @@
+import requests
+import logging
+import asyncio
+
 from aiohttp import web
 from os import environ as ENV
 from muswarmadmin.main import app
 from time import sleep
 
-import requests
-import logging
-import asyncio
-
 logger = logging.getLogger(__name__)
 loop = asyncio.get_event_loop()
 
-try:
+
+def pollAccumulate(count=0):
+    """
+    Poll the database until it is ready to answer queries
+    """
+    if count >= int(ENV['POLL_RETRIES']):
+        return False
     payload = {'query': 'select distinct ?c where {[] a ?c } LIMIT 1'}
     url = ENV['MU_SPARQL_ENDPOINT']
-    getout = False
-    while not getout:
-        r = requests.get(url, params=payload)
-        if r.ok:
-            logger.info('SPARQL endpoint is ready')
-            getout = True
-        else:
-            logger.warn('SPARQL endpoint not yet ready')
-        sleep(2)
+    try:
+        requests.get(url, params=payload)
+    except requests.RequestException:
+        logger.warn('SPARQL endpoint not yet ready')
+        sleep(1)
```

Comment:
To avoid all services to connect to the same database at the same time, the delay should be a random value.